### PR TITLE
Add erf function

### PR DESCRIPTION
**Pull request by benjello**
This PR adds an erf function using scipy.

### CODE REVIEW

**Comment by gdementen** [comment]
I do not want to depend on scipy, so the function must be either optional (only available if scipy is installed) or computed via an expression like shown at http://stackoverflow.com/questions/457408/is-there-an-easily-available-implementation-of-erf-for-python, if the later approach gives sufficient precision.

**Comment by benjello** [comment]
@gdementen : what is your favorite implementation ?
I would choose an optional scipy as a standard practice but the function you gave may be faster
and sufficient for liam2 needs.

**Comment by gdementen** [comment]
I have no idea if what I linked to is precise enough. I am not really qualified to tell. I would accept either, so do what you prefer/think is best. The second option might be faster indeed thanks to numexpr/multi-threading, but that would need to be tested.

**Comment by gdementen** [comment]
I cannot find to code of the erf function in scipy, it seems like it links to a C implementation from some other lib, so it is unlikely we will get any faster, at least with a single thread.

**Comment by benjello** [comment]
There is a link to the implementation in C on [this page](https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.special.erf.html).

I will use the scipy version.

**Comment by benjello** [comment]
If the implementation suits you I can even write a line in the changelog ;-)

**Comment by benjello** [comment]
@gdementen : any chance this get merged soon ?

File: liam2/exprmisc.py [hunk 1]
```diff
@@ -160,6 +165,21 @@ def compute(self, context, expr):
 
     dtype = always(int)
 
+
+class Erf(FunctionExpr):
+    def compute(self, context, expr):
+        if scipy is None:
+            raise ImportError(
+                "Scipy was not succesfully imported",
```
Comment (gdementen) [inline liam2/exprmisc.py]: I would like a better message for people trying to use it without scipy installed. Something like: 
"Failed to import scipy, which is required for erf(). Please make sure scipy is installed and working."

**Comment by benjello** [comment]
Needs to be done:
 - [ ] changelog
 - [ ] note in documentation

**Comment by benjello** [comment]
@gdementen : rebased, doc and changelog added

**Comment by gdementen** [comment]
Thanks

**Pull request merged**